src/foam/nanos/crunch/JavaCrunchService.java

```diff
@@ -0,0 +1,140 @@
+package foam.nanos.crunch;
+
+import foam.core.X;
+import foam.dao.ArraySink;
+import foam.dao.DAO;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.auth.Subject;
+import foam.nanos.auth.User;
+import foam.nanos.logger.Logger;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import static foam.mlang.MLang.*;
+
+public class JavaCrunchService implements CrunchService {
```

Comment:
JavaCrunchService seems a bit odd lol. 

It's up to you but I would have called the implemented class "CrunchServiceInterface" and this class "CrunchService" - Since it is the logic of the "CrunchService".

Reply:
What if we added a NodeCrunchService, or an open source contributor wanted to add PythonCrunchService?

Reply:
CrunchService is fine for the interface. 
and we can change the server side (java only) implementation name later if we like/need. 
so leave for now.

Reply:
Renamed it to `ServerCrunchService` since I realized this name can be overloaded for other languages.